src/funmisc.c

```diff
@@ -1562,3 +1567,273 @@ FUNCTION(fun_benchmark)
+    return;
+  }
+
+  mush_strncpy(tbuf, args[0], sizeof tbuf);
+  s = strchr(tbuf, '/');
+  if (!s) {
+    safe_strl("#-1 I need to know what attribute to trigger.", 45, buff, bp);
+    return;
+  }
+  *(s++) = '\0';
+  upcasestr(s);
+
+  thing = noisy_match_result(executor, tbuf, NOTYPE, MAT_EVERYTHING);
+
+  if (thing == NOTHING) {
+    return;
+  }
+
+  if (!controls(executor, thing)) {
+    safe_strl("#-1 PERMISSION DENIED", 21, buff, bp);
+    return;
+  }
+  
+  if (!fetch_ufun_attrib(args[0], executor, &ufun, UFUN_DEFAULT))
+  {
+    safe_strl("#-1 Invalid attribute to call", 29, buff, bp);
+    return;
+  }
+
+  req = mush_malloc(sizeof *req, "urlreq");
+  req->enactor = enactor;
+  req->thing = thing;
+  req->attrname = mush_strdup(s, "urlreq.attrname");
+  req->body = sqlite3_str_new(NULL);
+  req->too_big = 0;
+  req->pe_regs = pe_regs_create(PE_REGS_ARG | PE_REGS_Q, "fun_http");
+  pe_regs_qcopy(req->pe_regs, pe_info->regvals);
+
+  handle = curl_easy_init();
+  curl_easy_setopt(handle, CURLOPT_PROTOCOLS,
+                   CURLPROTO_HTTP | CURLPROTO_HTTPS | CURLPROTO_DICT);
+  curl_easy_setopt(handle, CURLOPT_URL, args[2]);
+  curl_easy_setopt(handle, CURLOPT_VERBOSE, 0);
+  curl_easy_setopt(handle, CURLOPT_NOPROGRESS, 1);
+  curl_easy_setopt(handle, CURLOPT_NOSIGNAL, 1);
+  curl_easy_setopt(handle, CURLOPT_SOCKOPTFUNCTION, req_set_cloexec);
+  /* We use half the queue entry timer to ensure we set reasonable duration expectations, and deal with rounding. */
+  curl_easy_setopt(handle, CURLOPT_CONNECTTIMEOUT, (int)round((float)options.queue_entry_cpu_time / 2000.0));
+  curl_easy_setopt(handle, CURLOPT_TIMEOUT, (int)round((float)options.queue_entry_cpu_time / 2000.0));
+  curl_easy_setopt(handle, CURLOPT_USERAGENT, "PennMUSH/1.8");
+  curl_easy_setopt(handle, CURLOPT_WRITEFUNCTION, req_write_callback);
+  curl_easy_setopt(handle, CURLOPT_WRITEDATA, req);
+  curl_easy_setopt(handle, CURLOPT_FOLLOWLOCATION, 1);
+  curl_easy_setopt(handle, CURLOPT_MAXREDIRS, 5);
+
+  userpass = pe_regs_get(req->pe_regs, PE_REGS_Q, "userpass");
+  if (userpass) {
+    curl_easy_setopt(handle, CURLOPT_USERPWD, userpass);
+    curl_easy_setopt(handle, CURLOPT_HTTPAUTH, CURLAUTH_ANY);
+  }
+
+  if (post || put || del) {
+    bool postdata;
+    const char *contenttype;
+
+    if (post) {
+      curl_easy_setopt(handle, CURLOPT_POST, 1);
+    } else if (put) {
+      curl_easy_setopt(handle, CURLOPT_CUSTOMREQUEST, "PUT");
+    } else {
+      curl_easy_setopt(handle, CURLOPT_CUSTOMREQUEST, "DELETE");
+    }
+
+    postdata = (bool)*args[3];
+
+    contenttype = pe_regs_get(req->pe_regs, PE_REGS_Q, "content-type");
+    if (contenttype) {
+      char ct_header[BUFFER_LEN];
+      snprintf(ct_header, sizeof ct_header, "Content-Type: %s", contenttype);
+      headers = curl_slist_append(headers, ct_header);
+    }
+    if (contenttype && postdata &&
+        (strstr(contenttype, "charset=utf-8") ||
+         strstr(contenttype, "charset=UTF-8"))) {
```

Comment:
A case-insensitive comparison here may be better?

Reply:
Agreed. I may just enhance the code that I copied it from as well.